code/clue.js

```diff
@@ -1,59 +1,342 @@
+  name: 'Guest House',
+  image: 'https://www.thesun.co.uk/wp-content/uploads/2019/12/60a659b1-5983-43ae-a6c0-5a2624d0384bjpg-JS550206142.jpg'
+};
+
+const patio = {
+  name: 'Patio',
+  image: 'https://steveandamysly.tannerworld.com/databank/2009/image_clue_reinvention04.jpg'
+};
+
+// GROUPING OF ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
-]
+];
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
 
-const rooms = []
+const rooms = [ 
+  diningRoom, 
+  conservatory, 
+  kitchen, 
+  studyRoom, 
+  libraryRoom,
+  billiardRoom, 
+  lounge, 
+  ballRoom,
+  hall,
+  spa,
+  livingRoom,
+  observatory, 
+  theater, 
+  guestHouse, 
+  patio
+];
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY. With a killer, a weapon and a room. The values are set later down.
+let theMystery = { // Write mystery instead?
+ killer: '',
+ weapon: '',
+ room: ''
+};
+
+// Variables for the pictures 
+const killerImage = document.getElementById('killerImage');
+const weaponImage = document.getElementById('weaponImage');
+const roomImage = document.getElementById('roomImage');
+
+// PICK-KILLER-FUNCTION! This function will be invoked when you click on the killer card.
+const pickKiller = () => { // Shows loading animation
+  document.getElementById('loader1').style.display = 'block';
+  setTimeout(realPickKiller, 1000); // Delays the realPickKiller function and calls it after 1.5 sec.
+};
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+const realPickKiller = () => { // The REAL function getting the values for the suspects
+  document.getElementById('loader1').style.display = 'none';
+  document.getElementById('killerCard').removeEventListener('click', pickKiller); // So the loader doesn't pops up in next click
+ 
+  mystery.killer = randomSelector(suspects);  // This will randomly select a killer from the suspects and add that to the mystery object.
+
+  // This function does randomly select a favourite weapon.
+  shuffleFavouriteWeapon = () => {
+    mystery.favouriteWeapon = randomSelector(weapons); 
+  };
+  shuffleFavouriteWeapon();
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the different properties of the killer. 
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
```

Comment:
Is it possible to format(is this the correct word haha) the code to make it easier the read on the longer strings? I had the same problem on my cluedo-project.

Something like:
document.getElementById(
 'killerName'
 ).innerHTML = 
`${mystery.killer.firstName}
 ${mystery.killer.lastName}`;

Reply:
Good point Mats. A solution to this is to use variables:
```javascript
const killerNameString = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
const killerNameLabel =   document.getElementById('killerName');
killerNameLabel.innerHTML = killerNameString;
```

Or something similar with different naming.